jmx/plugins/org.jboss.tools.jmx.ui/src/org/jboss/tools/jmx/ui/internal/wizards/DefaultConnectionWizardPage.java

```diff
@@ -275,8 +303,27 @@ protected String getNextName() {
 			return Messages.DefaultConnectionWizardPage_Default_Name;
 	}
 
+	private Job validateJob;
+	protected void validateJob() {
+		if( validateJob != null ) {
+			validateJob.cancel();
+		}
+		validateJob = new Job("Validate Wizard"){ //$NON-NLS-1$
```

Comment:
not sure if it is due to the Job or the asyncExec but, i'm able to to change a value in the UI then click Finish and the changed value is not saved, it has kept the old one.
of course, you need to change and click Finish (or type enter) quite fast